src/python/pants/pantsd/service/scheduler_service.py

```diff
@@ -137,9 +132,15 @@ def warm_product_graph(self, spec_roots):
       self._logger.debug('graph len was {}, waiting for initial watchman event'.format(graph_len))
       self._watchman_is_running.wait()
 
+    session = self._graph_helper.new_session()
+
     with self.fork_lock:
-      self._graph_helper.warm_product_graph(spec_roots)
-      return self._graph_helper
+      target_roots = target_roots_calculator.create(
```

Comment:
Does this need to be inside the lock?

Reply:
Yes, it should be. Wasn't before, but should be.